mongo/src/main/java/com/redhat/lightblue/mongo/crud/BasicDocSaver.java

```diff
@@ -288,59 +293,35 @@ private void updateDocs(CRUDOperationContext ctx,
                         try {
                             ctx.getFactory().getInterceptors().callInterceptors(InterceptPoint.PRE_CRUD_UPDATE_DOC, ctx, doc.inputDoc);
                             merge.merge(doc.oldDoc, doc.newDoc);
-                            Translator.populateDocHiddenFields(doc.newDoc, md);
+                            Translator.populateCaseInsensitiveHiddenFields(doc.newDoc, md);
                             updateAttemptList.add(doc);
                         } catch (Exception e) {
                             doc.inputDoc.addError(Error.get("update", MongoCrudConstants.ERR_TRANSLATION_ERROR, e));
                         }
                     } else {
                         doc.inputDoc.addError(Error.get("update",
-                                CrudConstants.ERR_NO_FIELD_UPDATE_ACCESS, paths.toString()));
+                                                        CrudConstants.ERR_NO_FIELD_UPDATE_ACCESS, paths.toString()));
                     }
                 }
                 LOGGER.debug("After checks and merge, updating {} docs", updateAttemptList.size());
                 if (!updateAttemptList.isEmpty()) {
-                    BulkWriteOperation bw = collection.initializeUnorderedBulkOperation();
+                    BulkUpdateCtx updateCtx=new BulkUpdateCtx();
+                    updateCtx.reset(collection);
```

Comment:
Done.

On Tue, Aug 9, 2016 at 12:42 PM, Alec Henninger <notifications@github.com>
wrote:

> In mongo/src/main/java/com/redhat/lightblue/mongo/crud/BasicDocSaver.java
> <https://github.com/lightblue-platform/lightblue-mongo/pull/299#discussion_r74119889>
> :
>
> >                      }
> >                  }
> >                  LOGGER.debug("After checks and merge, updating {} docs", updateAttemptList.size());
> >                  if (!updateAttemptList.isEmpty()) {
> > -                    BulkWriteOperation bw = collection.initializeUnorderedBulkOperation();
> > +                    BulkUpdateCtx updateCtx=new BulkUpdateCtx();
> > +                    updateCtx.reset(collection);
>
> I guess if you did that, it might also make sense to remove collection
> from the execute param. Is there a reason this is a param? I would think to
> correctly find any errors, it would have to be the same collection that was
> passed to reset, anyway? So maybe passing in constructor helps avoid a
> class of bugs if I understand correctly.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/lightblue-platform/lightblue-mongo/pull/299/files/3215df526dc8cf645462d5147d4ecfe0db012c92#r74119889>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADgDDR7T7HgiEc0qAlD6IsHRe_nuc2Lrks5qeMoRgaJpZM4JfOdd>
> .
>